File: kata-others/kata-ruby/a_case_of_snakes.rb
STDIN.each do |line|
  prefix = line.index(/[A-Z]/)
  puts prefix < 4 ? 
    line.gsub(/[A-Z]/) { |char| '_' + char.downcase }[prefix+1..] : 
    line.gsub(/[A-Z]/) { |char| '_' + char.downcase }
end

File: kata-others/kata-elixir/a_case_of_snakes.exs
IO.stream(:stdio, :line)
|> Enum.each(fn line ->
  snake = String.replace(line, ~r/[A-Z]/, &("_" <> String.downcase(&1))) |> String.trim()

  parts = String.split(snake, "_")
  [prefix | rest_list] = parts

  if(length(parts) > 2 && String.length(prefix) < 4, do: Enum.join(rest_list, "_"), else: snake)
  |> IO.puts()
end)
